stacks_queues/queue.py

```diff
@@ -1,61 +1,91 @@
 
 INITIAL_QUEUE_SIZE = 20
 
+
 class QueueFullException(Exception):
     pass
 
+
 class QueueEmptyException(Exception):
     pass
 
+
 class Queue:
 
     def __init__(self):
         self.store = [None] * INITIAL_QUEUE_SIZE
         self.buffer_size = INITIAL_QUEUE_SIZE
-        self.front = -1
+        self.front = -1  # to show there is nothing in the queue yet
         self.rear = -1
         self.size = 0
-      
 
     def enqueue(self, element):
         """ Adds an element to the Queue
             Raises a QueueFullException if all elements
             In the store are occupied
             returns None
         """
-        pass
+        if self.front == -1:
+            self.front = 0
+            self.rear = 0
+        elif self.size == self.buffer_size:
+            raise QueueFullException('Queue is full')
+
+        self.store[self.rear] = element
+        self.rear = (self.rear + 1) % self.buffer_size
+        self.size += 1
 
     def dequeue(self):
         """ Removes and returns an element from the Queue
             Raises a QueueEmptyException if 
             The Queue is empty.
         """
-        pass
+        if self.size == 0:
+            raise QueueEmptyException('Queue is empty')
+
+        head = self.front
+        element = self.store[head]
+
+        for i in range(self.size):
+            self.store[i] = self.store[i+1]
 
-    def front(self):
+        self.rear -= 1
+        self.size -= 1
```

Comment:
When we get the thing at the front, rather than shifting everything forward (O(n)), we can move where the front of the queue is in the circular buffer (O(1))
```suggestion
        self.front = (self.front + 1) % self.buffer_size
        self.size -= 1
```